Jordan

Jmgiacone opened the pull request.

Jmgiacone committed:
Clean up the main wordle gameplay loop as well as tweak the scoring algorithm

Jmgiacone committed:
Remove words that are not accepted by the wordle website

AdamNEvans committed:
Merge pull request #1 from Jmgiacone/wordle-solver-interface

Clean up the main wordle gameplay loop as well as tweak the scoring algorithm

Jmgiacone committed:
Put back the loadWords2(...) method

Jmgiacone committed:
To support the enablePrints param for the NoDupesSolver, default the initialize(...) to initialize(words, true)

Jmgiacone committed:
Rename stringRepresentation and associated getters/setters to charRepresentation in the LetterComparisonResult enum

Jmgiacone committed:
Switch back to the 5-letter wordle text file

Jmgiacone committed:
Merge branch 'master' into jordan

AdamNEvans commented on Utilities.java:
```diff
@@ -1,12 +1,33 @@
+		return charRepresentation;
+	}
+
+	public static Optional<LetterComparisonResult> fromStringRepresentation(char stringRepresentation)
```
forgot to change the function and parameter names

AdamNEvans replied:
still need to do this

AdamNEvans commented on Utilities.java:
```diff
@@ -92,6 +113,14 @@
 		return results;
 	}
 
+	public static String stringFromResults(LetterComparisonResult[] result)
```
Wondered if you'd add this function. Thanks

AdamNEvans replied:
Still need to do this

AdamNEvans commented on Wordle.java:
```diff
@@ -140,43 +146,29 @@ public static WordleSolver getSolver(Scanner in) throws Exception
 		}
 
 		// initialize the chosen solver
-		if (choice == 0)
-		{
-			NoDupesSolver solver = (NoDupesSolver)solvers[choice];
-			solver.initialize(loadWords(new File("words5.txt")), true);
-		}
-
-		return solvers[choice];
```
Why did you remove this only to add the same thing but generalized so that all solvers have to take the same data? I chose to do it this way so that solvers that didn't need to do much work could get away with it. If we end up adding code that runs a bunch of simulated games, we want the solver initializations to be fast (no unnecessary file loading for unused parameters), or at least that was my logic.

Jmgiacone committed:
Remove initialize() from WordleSolver interface

Add in a switch statement to switch on the choice from user input.
Then do all the implementation-specific initialization

AdamNEvans reviewed (commented):
you didn't address the comment on line 25 of Utilities.java

Jmgiacone committed:
Address review comments

Refactor the fromStringRepresentation(...) method to
fromCharRepresentation(...), chang the charRepresentation of UNKNOWN
to '@', and refactor the resultsFromString(...) method to use streams
instead of a for-loop
